texar/modules/pretrained/xlnet_model_utils.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Model Utils of XLNet Modules.
+
+Adapted from
+https://github.com/zihangdai/xlnet/blob/master/modeling.py
+"""
+
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from texar.core import get_layer
+from texar.module_base import ModuleBase
+
+__all__ = [
+    "PositionWiseFF",
+    "RelativeMultiheadAttention",
+    "RelativePositionalEncoding",
```

Comment:
These modules, including `Relative*` should be made texar modules, as discussed in https://github.com/asyml/texar-pytorch/pull/107

This is of lower priority though (lower than simplifying the xlnet example). We can probably open an issue first